### PR TITLE
ci: new step to check build sanity vs cosmos-sdk master

### PR DESCRIPTION
**Pull request by alessio**
Testbuild PRs against cosmos-sdk's current master, if applicable.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Make this an actions job if possible.

**Comment by alessio** [comment]
@alexanderbez makes sense, will do

**Comment by alessio** [comment]
@alexanderbez done

File: .github/workflows/build-latest-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - uses: rokroskar/workflow-run-cleanup-action@master
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
+
+  build-vs-cosmos-sdk-master:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'skip-tests')"
+    steps:
+      - uses: actions/setup-go@v1
```
Comment (tac0turtle) [inline .github/workflows/build-latest-sdk.yml:16]:
```suggestion
      - uses: actions/setup-go@v2-beta
```

File: .github/workflows/build-latest-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        id: go
+        with:
+          go-version: 1.14
+      - name: Setup env for GO
+        #  this is only used until the setup-go action is updated
+        run: |
+          echo "::set-env name=GOPATH::$(go env GOPATH)"
+          echo "::add-path::$(go env GOPATH)/bin"
+        shell: bash
```
Comment (tac0turtle) [inline .github/workflows/build-latest-sdk.yml:25]:
```suggestion
```

making these updates in the sdk as well. just found it. you can keep `with: <version>` if you would like to pin to a version

**Review by tac0turtle — APPROVED** [review]
LGTM

**Comment by alexanderbez** [comment]
Looks good but the job is failing.

**Review by alexanderbez — APPROVED** [review]
Looks good but the job is failing.

**Comment by alessio** [comment]
@alexanderbez yeah it's failing cause we're waiting on https://github.com/cosmos/gaia/pull/288. No harm is done if we merge this in the meantime.

**Comment by alexanderbez** [comment]
Got it, then I leave it to you to merge when ready.

**Comment by jackzampolin** [comment]
This is great improvement

**Review by jgimeno — APPROVED** [review]

**Comment by tac0turtle** [comment]
what is the benefit of having gaia build against master of the sdk?

before it was due to not having gaiad & gaiacli but now the sdk has simd and simcli

**Comment by alessio** [comment]
> what is the benefit of having gaia build against master of the sdk?

To ensure that `gaia`'s master always builds against `cosmos-sdk`'s latest master branch

**Comment by tac0turtle** [comment]
and what is the benefit for gaia to always be up to date with master?

just for context on why im asking is when an external module (non-SDK) gets added to Gaia this will not be possible unless that module also keeps up with master

**Comment by alessio** [comment]
> this will not be possible unless that module also keeps up with master

This is not meant to become a blocking job for a PR to be merged. It just gives the PR author additional information re: whether their PR builds successfully against cosmos-sdk master

**Comment by tac0turtle** [comment]
should we just merge this and see what people think about it?

**Pull request merged**